python/singa/autograd.py

```diff
@@ -4698,6 +4698,103 @@ def cossim(a, b):
     return CosSim()(a, b)[0]
 
 
+class Expand(Operator):
+    """
+    Init a expand operator
```

Comment:
You can explain what this operator does?
Does it reshape the tensor with more dimensions?

Reply:
what's the difference to reshape?

Reply:
I've added an example usage at the comment.

The `expand` duplicates the data along with the axis. It looks like `upsample` but still has some differences.
>
    Example usage::
    data = [[1.], [2.], [3.]]

    # dim_changed
    shape = [2, 1, 6]
    output = [[[1., 1., 1., 1., 1., 1.], 
               [2., 2., 2., 2., 2., 2.],
               [3., 3., 3., 3., 3., 3.]],
              [[1., 1., 1., 1., 1., 1.],
               [2., 2., 2., 2., 2., 2.],
               [3., 3., 3., 3., 3., 3.]]]

    # dim_unchanged
    shape = [3, 4]
    output = [[1., 1., 1., 1.],
              [2., 2., 2., 2.],
              [3., 3., 3., 3.]]
>

For upsample
>
    Example usage::
    data = [[[[1, 2],
              [3, 4],]]]

    # nearest
    scales = [1.0, 1.0, 2.0, 3.0]
    output = [[[[1, 1, 1, 2, 2, 2],
                [1, 1, 1, 2, 2, 2],
                [3, 3, 3, 4, 4, 4],
                [3, 3, 3, 4, 4, 4],]]]
>